optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -534,27 +549,54 @@ def pre_attn_forward(
+                    value_states,
+                    attention_mask,
+                    self.num_key_value_groups,
+                )
+
             if q_len == 1:
                 # next token
                 use_recompute = True if os.getenv("QUANT_CONFIG", "") else False
                 with ht.sdp_kernel(enable_recompute=use_recompute):
-                    attn_output = self.fused_scaled_dot_product_attention(
-                        query_states, key_states, value_states, attention_mask, 0.0, False, None, 'None'
-                    )
+                    if flash_attention_in_fp8 is True:
+                        attn_output = self.fused_scaled_dot_product_attention(
+                            query_states, key_states, value_states, attention_mask, 0.0, False, None, softmax_mode
+                        )
+                    else:
+                        attn_output = FusedSDPA.apply(
+                            query_states, key_states, value_states, attention_mask, 0.0, False, None, softmax_mode
+                        )
             else:
                 # first token
                 if flash_attention_causal_mask:
                     # causal masking on first token requires inputs to be of the same length
                     with ht.sdp_kernel(enable_recompute=flash_attention_recompute):
-                        attn_output = self.fused_scaled_dot_product_attention(
-                            query_states, key_states, value_states, None, 0.0, True, None, softmax_mode
-                        )
+                        if flash_attention_in_fp8 is True:
+                            attn_output = self.fused_scaled_dot_product_attention(
+                                query_states, key_states, value_states, None, 0.0, True, None, softmax_mode
+                            )
+                        else:
+                            attn_output = FusedSDPA.apply(
+                                query_states, key_states, value_states, None, 0.0, True, None, softmax_mode
+                            )
                 else:
                     with ht.sdp_kernel(enable_recompute=flash_attention_recompute):
-                        attn_output = self.fused_scaled_dot_product_attention(
-                            query_states, key_states, value_states, attention_mask, 0.0, False, None, softmax_mode
-                        )
-
+                        if flash_attention_in_fp8 is True:
+                            attn_output = self.fused_scaled_dot_product_attention(
+                                query_states, key_states, value_states, attention_mask, 0.0, False, None, softmax_mode
+                            )
+                        else:
+                            attn_output = FusedSDPA.apply(
+                                query_states, key_states, value_states, attention_mask, 0.0, False, None, softmax_mode
+                            )
+
+            if flash_attention_in_fp8 is True:
```

Comment:
please add 1 line comment on why this reshape is required for fp8 case

Reply:
It tuned out not to be necessary - removed this code fragment.